dali/pipeline/operators/paste/paste.cu

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/paste/paste.h"
+
+#include <utility>
+#include <vector>
+#include <algorithm>
+
+namespace dali {
+
+namespace {
+
+__global__
+__launch_bounds__(512, 1)
+void BatchedPaste(
+    const int N,
+    const int C,
+    const uint8* const __restrict__ fill_value,
+    const uint8* const * const __restrict__ in_batch,
+    uint8* const* const __restrict__ out_batch,
+    const int* const __restrict__ in_out_dims_paste_yx) {
+  const int n = blockIdx.x;
+
+  constexpr int nWaves = 16;
+  constexpr int blockSize = 512;
+  constexpr int nThreadsPerWave = blockSize / nWaves;
+  constexpr int MAX_C = 1024;
+
+  __shared__ uint8 rgb[MAX_C];
+  __shared__ int jump[MAX_C];
+  for (int i = threadIdx.x; i < C; ++i) {
```

Comment:
Maybe it is stupid question, but won't we end with threadIdx.x = 0 going through all values  0-C, threadIdx.x = 1 1-C and etc?
Maybe instead of ++i  there should be i+=blockDim.x?

Reply:
yup, of course